### PR TITLE
Add AsciiDoc plugin

### PR DESCRIPTION
**Pull request by olexii4**
### What does this PR do?

Add AsciiDoc plugin to this registry

https://github.com/eclipse/che/issues/14196

### CODE REVIEW

File: v3/plugins/joaompinto/asciidoctor-vscode/latest/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+publisher: joaompinto
+apiVersion: v2
+id: joaompinto.asciidoctor-vscode
+version: 2.7.6
```
Comment (AndrienkoAleksandr) [inline v3/plugins/joaompinto/asciidoctor-vscode/latest/meta.yaml]: I think this meta yaml will fail validation. Here should be version latest: https://github.com/eclipse/che-plugin-registry/blob/master/v3/plugins/eclipse/che-theia/latest/meta.yaml#L4

Comment (olexii4) [reply v3/plugins/joaompinto/asciidoctor-vscode/latest/meta.yaml]: Good catch. Thanks. Done

File: v3/plugins/joaompinto/asciidoctor-vscode/latest/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+id: joaompinto.asciidoctor-vscode
+version: latest
+type: VS Code extension
+name: AsciiDoctor
+title: AsciiDoctor
+description: AsciiDoc Support, and code browsing https://marketplace.visualstudio.com/items?itemName=joaompinto.asciidoctor-vscode
+icon: https://www.eclipse.org/che/images/logo-eclipseche.svg
+spec:
+  extensions:
+  - https://marketplace.visualstudio.com/_apis/public/gallery/publishers/joaompinto/vsextensions/asciidoctor-vscode/2.7.6/vspackage
```
Comment (amisevsk) [inline v3/plugins/joaompinto/asciidoctor-vscode/latest/meta.yaml]: Is the version here intentionally different? In general, the `latest` version of a plugin is just a bit of an ugly workaround; the `latest` `meta.yaml` should be a copy of the newest version in the registry with `version: latest`.

**Comment by amisevsk** [comment]
There's also an issue with there the metas are located; we still require `meta.yaml` to be in the `v3/plugins/publisher/name/version/` folder, so they should be in `v3/plugins/joaompinto/AsciiDoctor/latest`

File: v3/plugins/joaompinto/asciidoctor-vscode/latest/meta.yaml (hunk 2 above)
Comment (olexii4) [reply v3/plugins/joaompinto/asciidoctor-vscode/latest/meta.yaml]: done

**Comment by olexii4** [comment]
@amisevsk I have fixed

File: v3/plugins/joaompinto/asciidoctor-vscode/2.7.5/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+id: joaompinto.asciidoctor-vscode
+version: 2.7.5
+type: VS Code extension
+name: asciidoctor-vscode
+title: AsciiDoctor
+description: AsciiDoc Support, and code browsing https://marketplace.visualstudio.com/items?itemName=joaompinto.asciidoctor-vscode
+icon: https://www.eclipse.org/che/images/logo-eclipseche.svg
+spec:
+  extensions:
+  - https://marketplace.visualstudio.com/_apis/public/gallery/publishers/joaompinto/vsextensions/asciidoctor-vscode/2.7.5/vspackage
```
Comment (amisevsk) [inline v3/plugins/joaompinto/asciidoctor-vscode/2.7.5/meta.yaml]: Is there any reason to have the `2.7.5` version? Once it's in the plugin registry we don't have a process for deprecating and removing it. Unless there are essential compatability/feature differences between the two, I vote we remove `2.7.5`.

File: v3/plugins/joaompinto/asciidoctor-vscode/latest/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+publisher: joaompinto
+apiVersion: v2
+id: joaompinto.asciidoctor-vscode
+version: latest
+type: VS Code extension
+name: asciidoctor-vscode
```
Comment (amisevsk) [inline v3/plugins/joaompinto/asciidoctor-vscode/latest/meta.yaml]: The other vscode plugins have name formatted `vscode-<name>`. Could we update this to be `vscode-asciidoctor`?

File: v3/plugins/joaompinto/asciidoctor-vscode/latest/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+publisher: joaompinto
+apiVersion: v2
+id: joaompinto.asciidoctor-vscode
```
Comment (amisevsk) [inline v3/plugins/joaompinto/asciidoctor-vscode/latest/meta.yaml]: `id` shouldn't be included in the meta.yaml (`id` for this plugin would be `joaompinto/asciidoctor-vscode/latest`)

File: v3/plugins/joaompinto/asciidoctor-vscode/latest/meta.yaml (hunk 4 above)
Comment (olexii4) [reply v3/plugins/joaompinto/asciidoctor-vscode/latest/meta.yaml]: ok. Done

File: v3/plugins/joaompinto/asciidoctor-vscode/2.7.5/meta.yaml (hunk 3 above)
Comment (olexii4) [reply v3/plugins/joaompinto/asciidoctor-vscode/2.7.5/meta.yaml]: ok. Done

File: v3/plugins/joaompinto/asciidoctor-vscode/latest/meta.yaml (hunk 5 above)
Comment (olexii4) [reply v3/plugins/joaompinto/asciidoctor-vscode/latest/meta.yaml]: Done. Thanks.

**Comment by olexii4** [comment]
@amisevsk I have fixed

File: v3/plugins/joaompinto/vscode-asciidoctor/2.7.6/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+publisher: joaompinto
+apiVersion: v2
+version: 2.7.6
+type: VS Code extension
+name: vscode-asciidoctor
+displayName: AsciiDoc support
+title: AsciiDoctor Plugin.
+description: This extension provides a live preview, syntax highlighting and snippets for the AsciiDoc format using Asciidoctor flavor.
+icon: https://www.eclipse.org/che/images/logo-eclipseche.svg
```
Comment (vitaliy-guliy) [inline v3/plugins/joaompinto/vscode-asciidoctor/2.7.6/meta.yaml]: Change on https://raw.githubusercontent.com/asciidoctor/asciidoctor-vscode/v2.7.6/images/icon.png

File: v3/plugins/joaompinto/vscode-asciidoctor/latest/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+publisher: joaompinto
+apiVersion: v2
+version: latest
+type: VS Code extension
+name: vscode-asciidoctor
+displayName: AsciiDoc support
+title: AsciiDoctor Plugin.
+description: This extension provides a live preview, syntax highlighting and snippets for the AsciiDoc format using Asciidoctor flavor.
+icon: https://www.eclipse.org/che/images/logo-eclipseche.svg
```
Comment (vitaliy-guliy) [inline v3/plugins/joaompinto/vscode-asciidoctor/latest/meta.yaml]: Change on https://raw.githubusercontent.com/asciidoctor/asciidoctor-vscode/v2.7.6/images/icon.png as 2.7.6 is ised for latest

Comment (olexii4) [reply v3/plugins/joaompinto/vscode-asciidoctor/latest/meta.yaml]: Wrong icon type

**Comment by olexii4** [comment]
@vitaliy-guliy  '*.png' is wrong icon type
Screenshot from 2019-09-06 18-02-42

**Comment by olexii4** [comment]
Sorry. This PR was rebased and need rereview.

**Pull request merged**